### ISSUE TITLE
Layer - Discuss, then implement modal dialog accessibility best practices

### ISSUE THREAD

**Issue by halocline** [grommet/hpe-design-system#2397]
Follow on to https://github.com/grommet/hpe-design-system/issues/2332

Related to https://github.com/grommet/hpe-design-system/pull/2391#pullrequestreview-893939160

Add example where the content highlights the accessibility aspects such as where focus should be placed, proper roles, announcements, etc.

We have a decent start here: https://design-system.hpe.design/components/layer#accessibility, but haven't implemented consistently.

Determine:
- [ ] Where focus should be placed upon opening
- [ ] Role
- [ ] Screen reader dialog/announcement

Implementation
- [ ] Add example where the content highlights the accessibility aspects
- [ ] Accompanying content should explain what is happening and why

What opportunities are there to make these best practices either defaults or incorporated into the development workflow?
- [ ] Any Grommet enhancements?
- [ ] This may be an opportunity for adding to our ESLint as part of this issue.

**Comment by taysea** [comment]
Added accessibility guidelines in latest documentation: https://design-system.hpe.design/components/layer?q=layer#accessibility
